File: tests/fixtures/sample.java
package com.example.service;

import java.util.List;
import java.util.Map;

// WHY: Sample Java fixture exercising all extraction targets for Phase 9 tests.

/**
 * Repository interface defining the data-access contract.
 * Every store implementation must satisfy this interface.
 */
public interface DataRepository {
    void save(String item);
    String load(String key);
}

/**
 * Status enum for entity lifecycle tracking.
 */
public enum EntityStatus {
    ACTIVE,
    INACTIVE,
    DELETED
}

/**
 * A simple Point record carrying x/y coordinates.
 */
public record GeoPoint(int x, int y) {}

/**
 * WHY: DataStore is the primary repository used by the service layer.
 * It wraps a simple in-memory map for demonstration purposes.
 */
// HACK: using Map instead of a real DB — replace before production
@SuppressWarnings("unchecked")
@Service
public class DataStore implements DataRepository {

    private final Map<String, String> store;

    /**
     * Constructs a new DataStore backed by the given map.
     *
     * @param store the backing map (must not be null)
     */
    public DataStore(Map<String, String> store) {
        this.store = store;
        // NOTE: init() must be called to warm up the cache
        init();
    }

    /**
     * Persists the item under its key.
     *
     * @param item the value to store
     */
    @Override
    public void save(String item) {
        persist(item);
    }

    /**
     * Retrieves an item by key, returning null if absent.
     */
    @Override
    public String load(String key) {
        return retrieve(key);
    }

    // Private helpers below — not part of the public API.

    private void init() {
        this.store.clear();
    }

    private void persist(String item) {
        this.store.put(item, item);
    }

    private String retrieve(String key) {
        return this.store.get(key);
    }
}

File: tests/fixtures/sample.cs
// WHY: Sample C# fixture exercising all extraction targets for Phase 9 tests.
using System;
using System.Collections.Generic;

namespace SampleApp.Services
{
    /// <summary>
    /// Defines the contract for storage operations.
    /// </summary>
    public interface IDataRepository
    {
        void Save(string item);
        string Load(string key);
    }

    /// <summary>
    /// Lifecycle status values for tracked entities.
    /// </summary>
    public enum EntityStatus
    {
        Active,
        Inactive,
        Deleted
    }

    /// <summary>
    /// Lightweight type alias for callback notification.
    /// </summary>
    public delegate void NotifyCallback(string message);

    /// <summary>
    /// WHY: DataStore is the canonical repository implementation used by the service.
    /// Wraps a Dictionary for simplicity; swap for a real DB in production.
    /// </summary>
    // HACK: Dictionary used instead of real persistence — replace before go-live
    [Serializable]
    [DataContract]
    public class DataStore : IDataRepository
    {
        private readonly Dictionary<string, string> _store;

        /// <summary>
        /// Constructs a new DataStore.
        /// </summary>
        // NOTE: The backing dictionary is injected to allow test doubles.
        public DataStore(Dictionary<string, string> store)
        {
            _store = store;
            Initialize();
        }

        /// <summary>Persists the item using its value as key.</summary>
        public void Save(string item)
        {
            Persist(item);
        }

        /// <summary>Retrieves an item by key, or null if absent.</summary>
        public string Load(string key)
        {
            return Retrieve(key);
        }

        private void Initialize()
        {
            _store.Clear();
        }

        private void Persist(string item)
        {
            _store[item] = item;
        }

        private string Retrieve(string key)
        {
            return _store.ContainsKey(key) ? _store[key] : null;
        }
    }

    /// <summary>
    /// Read-only view over the repository.
    /// </summary>
    public struct DataView
    {
        public string Name { get; }

        public DataView(string name)
        {
            Name = name;
        }
    }
}
